session.go

```diff
@@ -220,6 +221,24 @@ func (s *Session) AcceptStream() (*Stream, error) {
 	}
 }
 
+// AcceptStream is used to block until the next available stream
+// is ready to be accepted. When the context is canceled, the session
+// is automatically closed.
+func (s *Session) AcceptStreamWithContext(ctx context.Context) (*Stream, error) {
+	select {
+	case <-ctx.Done():
+		s.Close()
```

Comment:
Do we really want this behavior? It seems like the outer side should have a `defer s.Close` anyways, and I can imagine wanting to perhaps timeout the accept, do some things, and then try again.